sql/catalyst/src/main/scala/org/apache/spark/sql/types/StructType.scala

```diff
@@ -301,8 +300,8 @@ case class StructType(fields: Array[StructField]) extends DataType with Seq[Stru
   def fieldIndex(name: String): Int = {
     nameToIndex.getOrElse(name,
       throw new IllegalArgumentException(
-        s"""Field "$name" does not exist.
-           |Available fields: ${fieldNames.mkString(", ")}""".stripMargin))
+        s""""$name" does not exist.
+           |Available: ${fieldNames.mkString("\"", "\", \"", "\"")}""".stripMargin))
```

Comment:
Can we keep the original form instead of adding additional `"` here?
```scala
scala> val s = org.apache.spark.sql.types.StructType.fromDDL("`\"` INT")
s: org.apache.spark.sql.types.StructType = StructType(StructField(",IntegerType,true))

scala> s("a")
java.lang.IllegalArgumentException: "a" does not exist.
Available: """
```

Reply:
Sure, I'll remove quotes around all versions of this exception.